src/common/libschedutil/ops.c

```diff
@@ -197,49 +251,47 @@ static void service_unregister (flux_t *h)
     flux_future_destroy (f);
 }
 
-struct ops_context *schedutil_ops_register (flux_t *h,
-                                            op_alloc_f *alloc_cb,
-                                            op_free_f *free_cb,
-                                            op_exception_f *exception_cb,
-                                            void *arg)
+int schedutil_ops_register (schedutil_t *s,
+                            op_alloc_f *alloc_cb,
+                            op_free_f *free_cb,
+                            op_exception_f *exception_cb,
+                            void *arg)
 {
     struct ops_context *ctx;
+    flux_t *h = schedutil_get_flux (s);
 
-    if (!h || !alloc_cb || !free_cb || !exception_cb) {
+    if (!s || !alloc_cb || !free_cb || !exception_cb) {
         errno = EINVAL;
-        return NULL;
+        return -1;
     }
     if (!(ctx = calloc (1, sizeof (*ctx))))
-        return NULL;
+        return -1;
     if (service_register (h) < 0)
         goto error;
     ctx->h = h;
     ctx->alloc_cb = alloc_cb;
     ctx->free_cb = free_cb;
     ctx->exception_cb = exception_cb;
     ctx->arg = arg;
-
-    if (flux_msg_handler_addvec (h, htab, ctx, &ctx->handlers) < 0)
+    schedutil_aux_set (s, "ops_ctx", ctx, (flux_free_f)schedutil_ops_destroy);
```

Comment:
Would it be cleaner if the schedutil "ops" were part of `schedutil_t` instead of stored in the aux hash. This might obviate the need for `schedutil_aux_get/set` (though I guess that might be handy for schedulers), and would remove the extra `malloc` here.

Not a big deal overall, though.

Reply:
Yeah. I waffled on that.  I originally just had the `ops` as part of the struct, but I was thinking that if we want to keep the `schedutil_t` definition out of the header, I would need to add a `schedutil_ops_get` function anyway.  Which brings up the question, how important is it to keep the `schedutil_t` definition out of the header?

Reply:
Added in https://github.com/flux-framework/flux-core/pull/2226/commits/73cefce93334e32378af106bda2017d30d1caa23